integration/oasis/stationfinder/low_energy_location_station_finder_test.go

```diff
@@ -0,0 +1,83 @@
+package stationfinder
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+)
+
+func createMockLowEnergyLocationStationFinder1() *lowEnergyLocationStationFinder {
+	obj := &lowEnergyLocationStationFinder{
+		location: nil,
+		bf: &basicFinder{
+			tnSearchConnector: nil,
+			searchResp:        mockSearchResponse1,
+			searchRespLock:    &sync.RWMutex{},
+		},
+	}
+	return obj
+}
+
+func createMockLowEnergyLocationStationFinder2() *lowEnergyLocationStationFinder {
+	obj := &lowEnergyLocationStationFinder{
+		location: nil,
+		bf: &basicFinder{
+			tnSearchConnector: nil,
+			searchResp:        mockSearchResponse2,
+			searchRespLock:    &sync.RWMutex{},
+		},
+	}
+	return obj
+}
+
+func createMockLowEnergyLocationStationFinder3() *lowEnergyLocationStationFinder {
+	obj := &lowEnergyLocationStationFinder{
+		location: nil,
+		bf: &basicFinder{
+			tnSearchConnector: nil,
+			searchResp:        mockSearchResponse3,
+			searchRespLock:    &sync.RWMutex{},
+		},
+	}
+	return obj
+}
+
+func TestLowEnergyLocationStationFinderIterator1(t *testing.T) {
+	sf := createMockLowEnergyLocationStationFinder1()
+	c := sf.iterateNearbyStations()
+	var r []ChargeStationInfo
+	go func() {
+		for item := range c {
+			r = append(r, item)
+		}
+		if !reflect.DeepEqual(r, mockChargeStationInfo1) {
+			t.Errorf("expect %v but got %v", mockChargeStationInfo1, r)
+		}
+	}()
+}
+func TestLowEnergyLocationStationFinderIterator2(t *testing.T) {
+	sf := createMockLowEnergyLocationStationFinder2()
+	c := sf.iterateNearbyStations()
+	var r []ChargeStationInfo
+	go func() {
```

Comment:
Why run test in `go func()`?

Reply:
Reply this message two days ago, but seems it didn't shows on your side. 

When design the functionality of `iterator`, I want to make which could support running algorithms in individual go-routine, that's why I made those test cases run in go func()

For the `base iterator`, it tested which in same `go-routine` and in different one.

Reply:
But maybe it's better to wait for the `go func()` done? Otherwise the `Testxxx` functions may report ok even its internal process may not finish yet(I guess).

Reply:
Yes, you are right.  Should wait for go routine finished.
Fixed in next pull: https://github.com/Telenav/osrm-backend/pull/189
With this commit: https://github.com/Telenav/osrm-backend/pull/189/commits/4263788d4032a51c60eb8a2c242d7835ac45d63b

Reply:
OK.